### PR TITLE
New HTTP Client and Misc Updates

### PR DESCRIPTION
**Pull request by kevinschoon**
Hi -- 

I have made a few minor updates to Qocache and started a few larger ones. This PR isn't quite finished yet but I wanted to open it and solicit some feedback.

#### Updates
* Introduced an HTTP client that allows reading/writing remote QFrames in the same style as the QFrame library.
* Added logging utilities with Uber's excellent [zap](https://github.com/uber-go/zap).
* Made the `query.Query` type public and added some functional options.

Thanks!

### CODE REVIEW

File: http/http.go [hunk 1]
```diff
@@ -364,12 +365,12 @@ func (a *application) status(w http.ResponseWriter, r *http.Request) {
 func Application(conf config.Config) *mux.Router {
 	c := cache.New(conf.Size, time.Duration(conf.Age)*time.Second)
 	s := statistics.New(c, conf.StatisticsBufferSize)
-	app := application{cache: c, stats: s}
+	app := application{cache: c, stats: s, log: config.Logger()}
 	r := mux.NewRouter()
 
-	// Mount on both qcache and qocache for compatibility with qcache
-	mw := chainMiddleware(withLz4)
```
Comment (tobgu) [inline http/http.go:368]: Where did the lz4 support go?

**Comment by tobgu** [comment]
Out of curiosity, and to understand the background for the contributions you are making:
What use do you see for Qocache and more specifically, what are you planning to use it for?

Regarding the HTTP client, perhaps you've seen it but I once made a very basic client lib for qcache in Go, it's here: https://github.com/tobgu/go-qcache-client

File: http/http.go (hunk 1 above)
Comment (kevinschoon) [reply http/http.go:368]: It is still there: https://github.com/tobgu/qocache/pull/1/files#diff-8a7ba22c391ff4531ce3e0182819ed4fR371

**Comment by kevinschoon** [comment]
Thanks for mentioning the qcache-client, I had not seen that. Immediately my goal was just a bit of cleanup and the HTTP client. Long term (out of the scope of this PR and maybe even as a separate project) I am interested in generalizing qocache into a "dataframe server" with optional persistent backends for reading and writing. Specifically I'd be using it in a finance project but I imagine it could be useful elsewhere as well.

**Comment by kevinschoon** [comment]
Decided to go a different direction with this. Closing this PR but I'll leave my branch in case the changes are helpful.

Thank you!

**Pull request closed without merging**